### DISCUSSION TITLE
Differences between LYWSD03MMC  HW 1.5 1.7 2.0?

### DISCUSSION THREAD

**Discussion by KeinPlandavonviel** [pvvx/ATC_MiThermometer#494] [General]
Hello,

I received new LYWSD03MMC today. To my surprise it is probably older stock in HW 1.5 with sensor SHTC3/SHTV3 vs SHT4x with HW 1.7.
Are there any differences in measurement accuracy or power consumption?
What I have noticed are different sensor settings as default values for humidity.
Any clarification would be nice. :)

Have a nice day

---

Simply out of curiosity.

HW 1.7 according to the flasher, but HW 2.0 according to the inside (why?). With one chinese hair in the epoxy. :o)

![HW 2 0 front](https://github.com/pvvx/ATC_MiThermometer/assets/126298371/930c3cda-fcba-4235-8435-bd38097929a0)
![hw 2 0 back](https://github.com/pvvx/ATC_MiThermometer/assets/126298371/ffb0f0f0-b3bd-42c4-a202-a8bf023eafc4)

HW 1.5

![b1 5 front](https://github.com/pvvx/ATC_MiThermometer/assets/126298371/bc5f527a-3dd3-4c89-94bb-70b5af9da707)
![b1 5 back](https://github.com/pvvx/ATC_MiThermometer/assets/126298371/5116fd91-a39b-457e-9a25-d30769c99066)

**Comment by KeinPlandavonviel** [comment]
SHTV3 vs SHTV4

https://sensirion.com/media/documents/D74DF14C/657C2CF1/HT_Transition_Guide_SHTV3_SHTV4.pdf

**Comment by pvvx** [comment]
[README](https://github.com/pvvx/ATC_MiThermometer#chipset-lywsd03mmc-hwb14)

|HW | LCD I2C   addr | SHTxxx   I2C addr | Note |
|-- | -- | -- | -- |
|B1.4 | 0x3C | 0x70   (SHTC3) |  |
|B1.5 | UART | 0x70   (SHTC3) |  |
|B1.6 | UART | 0x44   (SHT4x) |  |
|B1.7 | 0x3C | 0x44   (SHT4x) | Test   original string HW |
|B1.9 | 0x3E | 0x44   (SHT4x) |   |
|B2.0 | 0x3C | 0x44   (SHT4x) | Test   original string HW |

_Version 1.7 or 2.0 is determined at first run by reading the HW line written in Flash.
Display matrices or controllers are different for all versions, except B1.7 = B2.0._

---

B1.5 has higher consumption due to the control of the screen chip via UART.
The SHTv3/C3 sensor consumes more than the SHT4x. But SHTv3 handles changes in temperature and humidity more quickly.

**Reply by KeinPlandavonviel** [reply]
Many thanks for the information!
I really should have found/seen the readme myself, sorry.
Do you also have photos of an open B1.7 version? Simply out of curiosity. Would it also be possible to activate the heater of the SHT4x sensors in a later version of your firmware? So that it can be activated manually via the TeLinkFlasher if necessary?
Perhaps this would improve the sensor in very humid or outdoor areas in continuous operation.
With what accuracy are °C/humidity currently recorded? I assume that this also has an influence on the power consumption?

![heater](https://github.com/pvvx/ATC_MiThermometer/assets/126298371/ca3a1b64-0b22-4f21-98ec-d3a4391290e2)


OT:
At the moment I am still looking for a good adapter/chipset for BT5 LR. Do you have any tips on this? 
I have read your information here that some chipsets such as RTL8761BU reduce the performance for LR.
Are there currently chipsets that do not do this and run under Linux/Debian/Raspbian?

**Reply by pvvx** [reply]
RTL8761BU is sufficient. You just shouldn’t require it to accept dozens of sensors.

> Are there currently chipsets that do not do this and run under Linux/Debian/Raspbian?

https://github.com/pvvx/ATC_MiThermometer/discussions/481

**Reply by KeinPlandavonviel** [reply]
My main concern was the range. As far as I understood, the range with RTL8761BU in long range mode was exactly the same as in normal mode, right? Or have I misunderstood?
I had hoped that there was a chipset with which you could really go much further in long range mode.

EDIT:
I seem to have completely misunderstood about the reception level and have now read more here.
Do I understand correctly that a chipset with -110dBm reception level receives signals better than a chipset with -99dBm reception level?
In simple terms. It can process even weaker signals that the other chipset already rejects or no longer recognizes?

**Reply by pvvx** [reply]
> In simple terms. It can process even weaker signals that the other chipset already rejects or no longer recognizes?

Yes

**Reply by KeinPlandavonviel** [reply]
I have now looked for some BT adapters and found some with RTL8761BW (same as BU?), but also BT 5.4 adapters with BR8654. I think the chip comes from http://en.barrot.com.cn/

Do you know if they run under Linux/Debian?

EDIT:
Nevermind, unfortunately it does not run under Linux.

**Reply by pvvx** [reply]
Barrot chips run Linux with new kernel (patches). But the software included in the chips has several bugs and is not suitable for use with BLE.
It is unknown how to change the firmware in BR8654.
For RTL chips, the firmware is loaded every time at startup and there are no problems with updating versions.

PS: There is no support for Bluetooth 5.0 and higher in user applications for Linux, Windows, and Apple.

**Comment by umetnic** [comment]
>  With one chinese hair in the epoxy. :o)
> 
@KeinPlandavonviel, Hey! I noticed your amazing macro shots and was really curious which device you used. Would really love to know!"

**Comment by KeinPlandavonviel** [comment]
@umetnic

Thanks, but to be honest these were pretty average shots. Just with my smartphone (Oneplus 7T Pro wide angle lens) and a “suitable” version of Gcam. You can take even better photos with it though.
![PXL_20240906_134105725 ](https://github.com/user-attachments/assets/e86ac1c8-2ded-4b37-b94d-0a6c69a90ea7)

**Reply by umetnic** [reply]
@KeinPlandavonviel Wow, that’s awesome! Thanks for sharing the details. My wife actually has the OnePlus 7T, and I can't wait for her to get home so I can try this out with her phone. Really excited to see what we can capture with the Gcam app. Thanks again for the inspiration!

**Reply by pvvx** [reply]
Any flatbed scanner:
![image](https://github.com/user-attachments/assets/f0534d77-91a2-49d8-82e0-79552de504d2)
100% geometry...

**Reply by KeinPlandavonviel** [reply]
Not if you use an Epson V200 Photo. xD But thanks for the suggestion! :) Very good picture - what kind of scanner was that?

**Reply by pvvx** [reply]
The oldest model was used specifically to show what can be obtained on any flatbed scanner. In this case, the CanoScan 4400F.
![image](https://github.com/user-attachments/assets/b4ad0ccc-3832-4b42-8153-d16f8f048271)

Scan Quality |  
-- | --
Scan Sensor | Linear CCD
Optical Resolution | 4800 dpi
Hardware Resolution | 4800 x 9600 dpi
Interpolated Resolution | 19,200 x 19,200 dpi
Color Depth | 48-Bit RGB (16-Bit gray scale)
Dynamic Range | Not Specified
Scan Area |  
Reflective Scan Area | 8.5" x 11.7"
Transparency Scan Area | 1.5" x 10"
Film Formats | 35mm Slides and Negatives
Performance |  
Preview Speed | 2.6 seconds
Scan Speed | Not Specified
Batch Scanning | 6 frames 35mm film or 4 frames 35mm slides
Automation | Film Automatic Retouching and Enhancement (FARE)
Connectivity |  
Interface | USB 2.0 (Hi-Speed)
System & Software Requirements | Windows System566MHz Pentium III equivalent or higherWindows 98SE, 2000, Me, XP128MB RAMUSB Port (1.1or 2.0)Macintosh SystemPowerPC G3, G4, G5Mac OS X (10.2.8 to 10.4.x)256MB RAMBuilt in USB port
Physical |  
Light Source | Cold Cathode Fluorescent Lamp
Front Panel Controls | Scan, Copy, E-Mail and expanded PDF functions
Power Requirements | 120V AC, 60Hz
Dimensions | 10.2 x 3.3 x 18.7"
Weight | 6.2 lbs

**Comment by KeinPlandavonviel** [comment]
@umetnic:

I hope the OT is ok for pvvx. You can get the appropriate config files for the 7T here: https://www.celsoazevedo.com/files/android/p/module-enhance-op7-gcam/
The basic version should simply match the Android version. I've done quite well with the BSG versions so far, but didn't have a config back then and fiddled around myself.

Sounds a bit strange, but the phone itself doesn't even matter that much. Even with a Note4x and Gcam, I was able to take incredibly good pictures - compared to the Xiaomi camera. Macro is especially good with the wide-angle lenses. Or with the telephoto lens and magnifying glass in Macgyver style. (Very steady hands required)

Have fun :-)
